src/main/java/com/linkedin/kmf/services/KafkaMetricsReporterService.java

```diff
@@ -0,0 +1,139 @@
+                                 new Properties());
+  }
+
+  @Override
+  public synchronized void start() {
+    _executor.scheduleAtFixedRate(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          reportMetrics();
+        } catch (Exception e) {
+          LOG.error(_name + "/KafkaMetricsReporterService failed to report metrics", e);
+        }
+      }
+    }, _reportIntervalSec, _reportIntervalSec, TimeUnit.SECONDS);
+    LOG.info("{}/KafkaMetricsReporterService started", _name);
+  }
+
+  @Override
+  public synchronized void stop() {
+    _executor.shutdown();
+    _producer.close();
+    LOG.info("{}/KafkaMetricsReporterService stopped", _name);
+  }
+
+  @Override
+  public boolean isRunning() {
+    return !_executor.isShutdown();
+  }
+
+  @Override
+  public void awaitShutdown() {
+    try {
+      _executor.awaitTermination(Integer.MAX_VALUE, TimeUnit.MILLISECONDS);
+    } catch (InterruptedException e) {
+      LOG.info("Thread interrupted when waiting for {}/KafkaMetricsReporterService to shutdown", _name);
+    }
+    LOG.info("{}/KafkaMetricsReporterService shutdown completed", _name);
+  }
+
+  private void initializeProducer() throws Exception {
+    Properties producerProps = new Properties();
+    producerProps.put(ProducerConfig.ACKS_CONFIG, "-1");
+    producerProps.put(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG, "20000");
+    producerProps.put(ProducerConfig.RETRIES_CONFIG, "3");
+    producerProps.put(ProducerConfig.MAX_BLOCK_MS_CONFIG, String.valueOf(Long.MAX_VALUE));
+    producerProps.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, "1");
+    producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer");
+    producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer");
+    producerProps.put(ProducerConfig.CLIENT_ID_CONFIG, METRICS_PRODUCER_ID);
+    producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, _brokerList);
+    _producer = new KafkaProducer<>(producerProps);
+  }
+
+  private void reportMetrics() {
+    Map<String, String> metrics = new HashMap<>();
+    for (String metricName : _metricsNames) {
+      String mbeanExpr = metricName.substring(0, metricName.lastIndexOf(":"));
+      String attributeExpr = metricName.substring(metricName.lastIndexOf(":") + 1);
+      List<MbeanAttributeValue> attributeValues = getMBeanAttributeValues(mbeanExpr, attributeExpr);
+      for (MbeanAttributeValue attributeValue : attributeValues) {
+        String metric = attributeValue.toString();
+        String key = metric.substring(0, metric.lastIndexOf("="));
+        String val = metric.substring(metric.lastIndexOf("=") + 1);
+        metrics.put(key, val);
+      }
+    }
+    try {
+      LOG.info("Kafka Metrics Reporter sending metrics = " + _parser.writerWithDefaultPrettyPrinter().writeValueAsString(metrics));
```

Comment:
Would it be better to make this a debug level logging? If user wants to see metrics value in console, maybe they can use DefaultMetricsReporterService?